OpenSim/Simulation/Model/ScapulothoracicJoint.cpp

```diff
@@ -0,0 +1,288 @@
+}
+
+//=============================================================================
+// CONSTRUCTION
+//=============================================================================
+//_____________________________________________________________________________
+/**
+ * Construct properties with their default values.
+ */
+void ScapulothoracicJoint::constructProperties()
+{
+    setAuthors("Ajay Seth");
+    constructProperty_thoracic_ellipsoid_radii_x_y_z(Vec3(NaN));
+    constructProperty_scapula_winging_axis_origin(Vector(2, 0.0));
+    constructProperty_scapula_winging_axis_direction(0.0);
+}
+
+
+//=============================================================================
+// SCALING
+//=============================================================================
+//_____________________________________________________________________________
+/**
+* Scale a joint based on XYZ scale factors for the bodies.
+*
+* @param aScaleSet Set of XYZ scale factors for the bodies.
+* @todo Need to scale transforms appropriately, given an arbitrary axis.
+*/
+void ScapulothoracicJoint::extendScale(const SimTK::State& s,
+                                       const ScaleSet& scaleSet)
+{
+    // Joint knows how to scale locations of the joint in parent and on the body
+    Super::extendScale(s, scaleSet);
+
+    const std::string& parentName = getParentFrame().getName();
+
+    // Scaling related to the parent body:
+    //
+    // Joint kinematics are scaled by the scale factors for the parent body,
+    // so get those body's factors
+    Vec3 scaleFactors(1.0);
+    for (int i = 0; i < scaleSet.getSize(); i++) {
+        const Scale &scale = scaleSet.get(i);
+
+        if (scale.getSegmentName() == parentName) {
+            scale.getScaleFactors(scaleFactors);
+            break;
+        }
+    }
+
+    // Scale the size of the mobilizer
+    for (int i=0; i<3; i++) {
+        upd_thoracic_ellipsoid_radii_x_y_z()[i] *= scaleFactors[i];
+    }
+}
+
+//=============================================================================
+// Simbody Model building.
+//=============================================================================
+//_____________________________________________________________________________
+void ScapulothoracicJoint::extendAddToSystem(SimTK::MultibodySystem& system) const
+{
+    Super::extendAddToSystem(system);
+
+    // Transform for ellipsoid joint frame in intermediate Scapula massless
+    // body frame.
+    //
+    // Oriented such that intermediate frame is aligned with scapula joint
+    // frame with respect to the scapula body frame as user specified by
+    // _location and _orientation
+    Transform ellipsoidJointFrameInIntermediate{
+        Rotation{
+            BodyRotationSequence,
+            0, XAxis,
+            0, YAxis,
+            Pi/2,
+            ZAxis},
+        Vec3(0.0)};
+
+    // Transform for Ellipsoid in parent body (Thorax).
```

Comment:
In larger functions, multi-step initialization algs are rolled up into standalone lambdas that are immediately called. This reduces the number of local variables in the function, which should hopefully make it easier to keep track of what's happening in the top-level func, and neatly groups bigger initialization steps into blocks /w a comment (e.g. all of the stuff below is *specifically* for initializing `ellipsoidJointFrameInThorax`).

Reply:
Main concern with use of lambdas is that our typical users are really comfortable with Matlab/python level syntax, very little C++ so we need to factor in the need for the code to stay readable/understandable for them.  This maybe ok in this case if it avoids code duplication but something to keep in mind going forward.